src/Compilers/CSharp/Test/Emit3/Attributes/AttributeTests_Embedded.cs

```diff
@@ -193,61 +195,400 @@ public static void Main()
-    {
-        // This should trigger generating another EmbeddedAttribute
-    }
-}";
+        [Theory, CombinatorialData]
+        public void EmbeddedAttributeFromSourceValidation_MissingEmbeddedAttributeApplicationGeneratesAttributeApplication(bool hasObsolete)
+        {
+            var code = $$"""
+                namespace Microsoft.CodeAnalysis
+                {
+                    {{(hasObsolete ? "[System.Obsolete]" : "")}}
+                    internal sealed class EmbeddedAttribute : System.Attribute
+                    {
+                        public EmbeddedAttribute() { }
+                    }
+                }
+                class Test
+                {
+                    public void M(in int p)
+                    {
+                        // This would trigger generating EmbeddedAttribute if we hadn't defined one
+                    }
+                }
+                """;
+
+            var comp = CreateCompilation(code, assemblyName: "testModule", targetFramework: TargetFramework.NetStandard20);
+            var embeddedAttribute = comp.Assembly.GetTypeByMetadataName(AttributeDescription.CodeAnalysisEmbeddedAttribute.FullName);
+            Assert.True(embeddedAttribute.HasCodeAnalysisEmbeddedAttribute);
+
+            CompileAndVerify(comp, symbolValidator: static module =>
+                {
+                    var embeddedAttribute = module.ContainingAssembly.GetTypeByMetadataName(AttributeDescription.CodeAnalysisEmbeddedAttribute.FullName);
+                    Assert.NotNull(embeddedAttribute);
+                    Assert.Equal(["Microsoft.CodeAnalysis.EmbeddedAttribute"], embeddedAttribute.GetAttributes().Where(a => a.AttributeClass.Name != "ObsoleteAttribute").Select(a => a.AttributeClass.ToTestDisplayString()));
+                })
+                .VerifyDiagnostics();
+        }
 
-            CreateCompilation(code, references: new[] { moduleRef }).VerifyEmitDiagnostics(
-                // error CS8004: Type 'EmbeddedAttribute' exported from module 'testModule.netmodule' conflicts with type declared in primary module of this assembly.
-                Diagnostic(ErrorCode.ERR_ExportedTypeConflictsWithDeclaration).WithArguments("Microsoft.CodeAnalysis.EmbeddedAttribute", "testModule.netmodule").WithLocation(1, 1));
+        [Fact]
+        public void EmbeddedAttributeFromSourceValidation_Generic()
+        {
+            var code = """
+                namespace Microsoft.CodeAnalysis
+                {
+                    [Embedded<int>]
+                    internal sealed class EmbeddedAttribute<T> : System.Attribute
+                    {
+                        public EmbeddedAttribute() { }
+                    }
+                }
+                class Test
+                {
+                    public void M(in int p)
+                    {
+                        // This would trigger generating another EmbeddedAttribute, if we hadn't already defined one
```

Comment:
>// This would trigger generating another EmbeddedAttribute, if we hadn't already defined one [](http://example.com/codeflow?start=24&length=92)

Should this be "This should trigger generating another EmbeddedAttribute"?